push-rewrite

```diff
@@ -67,9 +67,10 @@ def main():
     pull = find_pr_with_sha(api, remote)
     labels = labels_of_pull(pull)
     tests_disabled = "no-test" in labels or "[no-test]" in pull["title"]
-    # add no-test label to prevent tests bring triggered
+    # needs no-test label to prevent tests bring triggered
+    # we cannot set the label for ourselves without `repo` permission
     if not tests_disabled:
-        label(pull, {"labels": labels + ["no-test"]})
+        sys.exit("Please set the 'no-test' label on the PR before trying this")
```

Comment:
It could also just try label(), and intercept the IOError (or whatever it is), and fall back to the "please set"?

Just a suggestion, not a veto, I'm fine either way.

Reply:
I did that first, but it was messy and I wasn't sure what to write for the error message.  I started like "You need to add the `public_repo` scope on your ..." and realized that we are actively discouraging people from doing that these days.  That's when I changed tack.